types/enums/PlanningConstraints.ts

```diff
@@ -31,3 +33,40 @@ export const PlanningDesignations = {
   'road.classified': 'Classified Road',
   tpo: 'Tree Preservation Order (TPO) or zone',
 };
+
+type PlanningDesigationKeys = keyof typeof PlanningDesignations;
+
+type GenericPlanningDesignation<TKey extends PlanningDesigationKeys> = {
+  value: TKey;
+  description: (typeof PlanningDesignations)[TKey];
+};
+
+type PlanningDesignationMap = {
+  [K in PlanningDesigationKeys]: GenericPlanningDesignation<K>;
+};
+
+type BasePlanningDesignation =
+  PlanningDesignationMap[keyof PlanningDesignationMap];
+
+/**
+ * @description A planning designation that does not intersect with the proposed site, per the DE-9IM spatial relationship definition of intersects
+ */
+type NonIntersectingPlanningDesignation = {
+  intersects: false;
+} & BasePlanningDesignation;
+
+/**
+ * @description A planning designation that does intersect with the proposed site, per the DE-9IM spatial relationship definition of intersects
+ */
+type IntersectingPlanningDesignation = {
+  intersects: true;
+  entities: Entity[] | [];
```

Comment:
What's the reason this isn't strictly an `Entity[]`?

Reply:
This is a bit of a weird relic. 

You'll find it in existing examples like this (constraint values which don't map to a single Planning Data dataset etc): 
```
{
  value: 'designated',
  description: 'Designated land',
  intersects: true,
  entities: [],
},
```

I think it was a strategy to _require_ `entity` when intersects is `true`. But we could also make entity optional, then enforce `entities: Entity[]`. I'll revisit this one in another PR as a number of other examples will have to be updated!

Reply:
Thanks for the explanation 👍